File: backend/bibliotheque/src/main/java/com/upb/bibliotheque/entity/Utilisateur.java
package com.upb.bibliotheque.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDateTime;

@Entity
@Table(name = "utilisateurs")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "type_utilisateur")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Utilisateur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long utilisateurId;

    @Column(nullable = false, unique = true)
    private String identifiant;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String motDePasse;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private String prenom;

    private LocalDateTime dateCreation = LocalDateTime.now();
    private boolean actif = true;
}

File: shared/models/utilisateur.dart
class Utilisateur {
  final int? utilisateurId;
  final String identifiant;
  final String email;
  final String nom;
  final String prenom;
  final String? filiere;
  final String? anneeEtude;
  final int scoreReputation;
  final bool actif;
  final String typeUtilisateur;

  Utilisateur({
    this.utilisateurId,
    required this.identifiant,
    required this.email,
    required this.nom,
    required this.prenom,
    this.filiere,
    this.anneeEtude,
    this.scoreReputation = 0,
    this.actif = true,
    this.typeUtilisateur = 'ETUDIANT',
  });

  factory Utilisateur.fromJson(Map<String, dynamic> json) => Utilisateur(
    utilisateurId: json['utilisateurId'],
    identifiant: json['identifiant'] ?? '',
    email: json['email'] ?? '',
    nom: json['nom'] ?? '',
    prenom: json['prenom'] ?? '',
    filiere: json['filiere'],
    anneeEtude: json['anneeEtude'],
    scoreReputation: json['scoreReputation'] ?? 0,
    actif: json['actif'] ?? true,
    typeUtilisateur: json['type_utilisateur'] ?? 'ETUDIANT',
  );

  String get initiales =>
      '${prenom.isNotEmpty ? prenom[0] : ''}${nom.isNotEmpty ? nom[0] : ''}';
}